### PR TITLE
Fix datepicker date parsing when using a custom date format

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
Fixes https://github.com/valor-software/ngx-bootstrap/issues/3521

### CODE REVIEW

**Comment by valorkin** [comment]
Could you apply same change to date range input please

**Comment by FrancescoBorzi** [comment]
done

File: demo/src/app/components/+datepicker/demos/custom-format/date-picker-custom-format.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<div class="row">
+  <div class="col-xs-12 col-12 col-md-4 form-group">
+    <form [formGroup]="myForm">
+      <input
+        class="form-control"
+        [minDate]="minDate"
+        [maxDate]="maxDate"
+        #dp="bsDatepicker"
+        bsDatepicker
+        [formControlName]="'myDate'"
```
Comment (valorkin) [inline demo/src/app/components/+datepicker/demos/custom-format/date-picker-custom-format.html]: it's just a string setter `formControlName="myDate"`

**Review by valorkin — DISMISSED** [review]

File: demo/src/app/components/+datepicker/demos/custom-format/date-picker-custom-format.html (hunk 1 above)
Comment (FrancescoBorzi) [reply demo/src/app/components/+datepicker/demos/custom-format/date-picker-custom-format.html]: it's just a matter of codestyle, anyway I changed it as you suggested

**Comment by YevheniiaMazur** [comment]
tested, looks good

**Comment by olezt** [comment]
I think there is still an issue with this. Before the fix if dateInputFormat was 'DD-MM-YYYY', entering manually date 01-02-2018 would become 02-01-2018. Now it is not reversed on input but when opening the datepicker I still see the date reversed.

**Pull request merged**